### PR TITLE
updated

### PR DESCRIPTION
**Pull request by SAIKAT123GATE**

### CODE REVIEW

File: controllers/index.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+//common function
+function parseAndStringify(data) {
+    data = JSON.stringify(data);
+    data = JSON.parse(data);
+
+    return data;
+}
+
+//find All User
```
Comment (abhinit) [inline controllers/index.js:17]: Kindly remove this function, as it is not required

File: controllers/index.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        return upsertData
+    }
+    else {
+        var message = "User Already Exists";
+        return message;
+    }
+}
+
+//create & Update Transaction
+async function createTransaction(req){
```
Comment (abhinit) [inline controllers/index.js:43]: Please add an endpoint to insert multiple transactions, create a separate endpoint for insert and update

File: controllers/index.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
+var initModels = require("../models/init-models");
```
Comment (abhinit) [inline controllers/index.js:1]: Please create separate controllers for user and transactions (renamed from data)

File: models/User.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+module.exports = function(sequelize, DataTypes) {
+  return sequelize.define('User', {
+    UserID: {
+      autoIncrement: true,
+      type: DataTypes.INTEGER,
+      allowNull: false,
+      primaryKey: true
+    },
+    Email: {
+      type: DataTypes.STRING(45),
```
Comment (abhinit) [inline models/User.js:11]: Datatype for the Email should be varchar(254)

File: controllers/index.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    return data;
+}
+
+//find All User
+function findAllUser() {
+    return models.User.findAll({});
+}
+
+//get Particular User Details
+function getParticularUserDetailsById(UserId) {
```
Comment (abhinit) [inline controllers/index.js:23]: Let's not include 'data' and just a bank balance

File: package.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (abhinit) [inline package.json:1]: Kindly add npm scripts for running the server

File: routes/index.js [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+    }
+
+    // app.use(Middleware);
+    app.get('/', (req, res) => {
+        sendSuccess(res,"Welcome")
+    })
+
+    //get all user table
+    app.get("/alluser",Middleware,(req, res) => {
```
Comment (abhinit) [inline routes/index.js:33]: No need of this endpoint

File: routes/index.js [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+    app.put("/retail",Middleware, (req, res) => {
+        controller.createTransaction(req).then((response) => {
+            sendSuccess(res, response);
+        }).catch((error) => {
+            sendError(res, error);
+        })
+    })
+
+    //Delete Transactions
+    app.delete("/retail/delete",Middleware, (req, res) => {
```
Comment (abhinit) [inline routes/index.js:80]: No need of /delete in the endpoint, the request type 'delete' is enough to match this route

File: routes/index.js [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+async function handleCallback(responsePromise, res) {
+    responsePromise.then((response) => {
+        sendSuccess(res, response);
+    }).catch((error) => {
+        sendError(res, error)
+    });
+}
+
+
+function sendSuccess(res, response) {
```
Comment (abhinit) [inline routes/index.js:109]: Can we add the status code?

File: routes/index.js [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+    const result = {
+        "status": "ok",
+        "data": response
+    }
+    res.send(result)
+}
+
+function sendError(res, error) {
```
Comment (abhinit) [inline routes/index.js:117]: Can we add a status code?

**Review by abhinit — CHANGES_REQUESTED** [review]
Thanks a lot for the good work! I suggest you time yourself and only spend 30mins to 1hr on this max, please do only the needful.

**Comment by SAIKAT123GATE** [comment]
Done All Changes

**Comment by abhinit** [comment]
Thank you for the hard work, am going through the code. Will get back to you asap. Thanks.

**Pull request merged**